File: Stack/22. Generate Parentheses/sol.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:

        stack = []
        resut = []

        def backtrack(o, closed):
            if o == closed == n:
                res.append("".join(stack))
                return
            
            if o < n
                stack.append('(')
                backtrack(openN + 1, closedN)
                stack.pop()
            if closed < open;
            

File: LinkedList/138. Copy List with Random Pointer/sol.ts
function generateParenthesis(n: number): string[] {
  const stack: string[] = [];
  const res = [];
  const backtrack = (closeP: number, openP: number) => {
    if (stack.length === n * 2) res.push(stack.join(''));
    console.log(closeP, openP);
    if (openP <= n) {
      stack.push('(');
      backtrack(closeP, openP + 1);
      stack.pop();
    }
    if (closeP < openP) {
      stack.push(')');
      backtrack(closeP + 1, openP);
      stack.pop();
    }
  };
  backtrack(1, 1);
  return res;
}

generateParenthesis(2);
